stan/math/prim/fun/multiply.hpp

```diff
@@ -22,11 +22,48 @@ namespace math {
  */
 template <typename Mat, typename Scal, require_stan_scalar_t<Scal>* = nullptr,
           require_eigen_t<Mat>* = nullptr,
-          require_all_not_st_var<Scal, Mat>* = nullptr>
+          require_all_not_st_var<Scal, Mat>* = nullptr,
+          require_all_not_complex_t<Scal, value_type_t<Mat>>* = nullptr>
 inline auto multiply(const Mat& m, Scal c) {
   return c * m;
 }
 
+/**
+ * Multiply specialization for left or right hand side with a complex value type
```

Comment:
[optional]
I'd really like to see this doc be more user-facing than dev-facing.  The first sentence should always be formulated in terms of what the arguments are and return value is.  Specialization is a dev concept.  The way I'd write the doc for this is

```
* Return the product of the specified matrix and scalar, one of which must have a complex value type.
* The return type will be an expression template denoting a complex matrix.
*
* @tparam Mat type of matrix argument
* @tparam Scal type of scalar argument
* @param m matrix argument
* @param c scalar argument
* @return expression template evaluating to the product of the matrix and scalar arguments
```

Reply:
V reasonable and I like this wording a lot more. Fixed!